Userland/Libraries/LibWeb/SVG/SVGGraphicsElement.cpp

```diff
@@ -57,6 +57,23 @@ Optional<Gfx::PaintStyle const&> SVGGraphicsElement::fill_paint_style(SVGPaintCo
     return {};
 }
 
+Optional<Gfx::PaintStyle const&> SVGGraphicsElement::stroke_paint_style(SVGPaintContext const& paint_context) const
+{
+    // FIXME: This entire function is an ad-hoc hack:
+    if (!layout_node())
+        return {};
+    auto& stroke = layout_node()->computed_values().stroke();
+    if (!stroke.has_value() || !stroke->is_url())
+        return {};
+    auto& url = stroke->as_url();
+    auto gradient = document().get_element_by_id(url.fragment());
+    if (!gradient)
+        return {};
+    if (is<SVG::SVGGradientElement>(*gradient))
+        return static_cast<SVG::SVGGradientElement const&>(*gradient).to_gfx_paint_style(paint_context);
+    return {};
```

Comment:
Maybe factor this into a another function from that takes an `SVGPaint`, since it is pretty much identical for fills